core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -230,44 +238,65 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
-                        .get({
-                            itemsPerPage: 40,
-                            offset: offset.toString(),
-                            query: this.getESQuery(),
-                            sortField: sortField || 'modDate',
-                            sortOrder: sortOrderValue
-                        })
-                        .pipe(
-                            tapResponse(
-                                (items) => {
-                                    const dateFormattedPages = this.formatRelativeDates(
-                                        items.jsonObjectView.contentlets
-                                    );
-
-                                    let currentPages = this.get().pages.items;
-
-                                    if (currentPages.length === 0) {
-                                        currentPages = Array.from({ length: items.resultsSize });
-                                    }
-
-                                    Array.prototype.splice.apply(currentPages, [
-                                        ...[offset, 40],
-                                        ...dateFormattedPages
-                                    ]);
-
-                                    this.setPages(currentPages);
-                                },
-                                (error: HttpErrorResponse) => {
-                                    return this.httpErrorManagerService.handle(error);
-                                }
-                            )
-                        );
+                    return this.getPagesData(offset, sortOrderValue, sortField).pipe(
+                        tapResponse(
+                            (items) => {
+                                const formatedPages = this.formatPagesData(items, offset);
+                                this.setPages(formatedPages);
+                            },
+                            (error: HttpErrorResponse) => {
+                                return this.httpErrorManagerService.handle(error);
+                            }
+                        )
+                    );
                 })
             );
         }
     );
 
+    readonly getPagesRetry = this.effect((params$: Observable<{ offset: number }>) => {
+        return params$.pipe(
+            mergeMap((params) => {
+                const { offset } = params;
+                const sortOrderValue = this.getSortOrderValue();
+
+                let initialData = '';
+                let retries = 0;
+
+                return this.getPagesData(offset, sortOrderValue)
+                    .pipe(
+                        tap(
+                            (items) => {
+                                retries++;
+
+                                // First fetch to grab initial data to compare
+                                if (!initialData) {
+                                    initialData = JSON.stringify(items.jsonObjectView);
```

Comment:
the only way to be safe that I'll get a different set of data is by doing a pull from the endpoint, otherwise due to data transformations there could be errors.
Also it does not matter if an initial request is made, because due to the BE bug reindex bug, there always is going to be needed at least 3 or 4 request until the data gets updated from the DB